### PR TITLE
Add logging of IOMCU stats, raise threshold of failures for IOMCU to be unhealthy

### PR DESCRIPTION
**Pull request by WickedShell**
We fail ~0.6% of all writes to the IOMCU. This is an arming check, which leads to very brief transients of failing the arming checks, which it is possible to hit. This raises the threshold for the IOMCU to be considered unhealthy to be more then 5 failures with communicating with the IOMCU. My testing shows they accumulate at a rate of 2Hz normally, and we issue these at a high rate.

This also brings in a bunch of logging for the IOMCU library to allow us to have general insights to this. I had more logging during testing, but this is the minimum set I think that is currently helpful. I'm hoping @peterbarker has suggestions on where I can stick the call to log that would be common code, rather then copy pasting it. The logging code is technically a race condition as well, although I've currently taken the approach of ignoring that as it will be better by the next log call. With that being said if it bothers anyone for any reasons I can go back in and add the required locking.

### CODE REVIEW

File: ArduPlane/ArduPlane.cpp [hunk 1]
```diff
@@ -310,6 +310,12 @@ void Plane::one_second_loop()
     // indicates that the sensor or subsystem is present but not
     // functioning correctly
     gcs().update_sensor_status_flags();
+
+#if HAL_WITH_IO_MCU
```
Comment (peterbarker) [inline ArduPlane/ArduPlane.cpp]: It would be nice to be able to put this into IOMCU directly - otherwise we need this patch in the other vehicles.  There's no 1Hz loop but you could run a hal-millis-based timer in something which will always be called.

This would involve passing in a bitmask to the iomcu library at init.

File: ArduPlane/ArduPlane.cpp [hunk 2]
```diff
@@ -310,6 +310,12 @@ void Plane::one_second_loop()
...
+    if (should_log(MASK_LOG_PM)) {
```
Comment (peterbarker) [inline ArduPlane/ArduPlane.cpp]: This should probably be `MASK_LOG_RC` on some vehicles and `MASK_LOG_RCIN|MASK_LOG_RCOU` on others, I think?

File: libraries/AP_IOMCU/AP_IOMCU.cpp [hunk 3]
```diff
@@ -786,7 +801,7 @@ void AP_IOMCU::set_safety_mask(uint16_t chmask)
  */
 bool AP_IOMCU::healthy(void)
 {
-    return crc_is_ok && protocol_fail_count == 0;
+    return crc_is_ok && stats.protocol_fail_count > 5;
```
Comment (peterbarker) [inline libraries/AP_IOMCU/AP_IOMCU.cpp]: I wonder if this should be split into two PRs...

Comment (WickedShell) [reply libraries/AP_IOMCU/AP_IOMCU.cpp]: Feel free to cherry pick this part into a PR. It is needed to fix a 0.6% chance of failing to arm on QuadPlane (when everything is otherwise normal). I put it in this, because I didn't think this would be stalled on lots of changes to IOMCU :)

File: ArduPlane/ArduPlane.cpp (hunk 2 above)
Comment (WickedShell) [reply ArduPlane/ArduPlane.cpp]: I could argue either way. I selected PM, because this is basically failure monitoring of the communication health. I can be convinced otherwise :)

File: ArduPlane/ArduPlane.cpp (hunk 1 above)
Comment (WickedShell) [reply ArduPlane/ArduPlane.cpp]: Yeah, I really debated it, but at the implementation time I wasn't sold on if calling logger from other threads was safe. If it is, then it fixes the race condition I outlined above. If it's not then you could stall the IO comm link on accounting of logging, which seems like a bad thing (tm)

**Comment by tridge** [comment]
replaced by #12033

**Comment by tridge** [comment]
@WickedShell note that #12033 fixes the cause of the high protocol_fail_count errors
